packages/visx-vendor/scripts/buildVendor/utils.ts

```diff
@@ -0,0 +1,162 @@
+    (all, packageName) => {
+      const npmAlias = pkgToAliasMap[packageName];
+      const isTypeFile = packageName.includes('@types/');
+      const pkg: VendoredPkg = {
+        packageName,
+        npmAlias,
+        isTypeFile,
+        tsFileName: isTypeFile
+          ? `${ROOT_PATH}/${packageName.replace('@types/', '')}.d.ts`
+          : undefined,
+        esmFileName: `${ESM_PATH}/${packageName}.js`,
+        cjsFileName: `${CJS_PATH}/${packageName}.js`,
+        indexFileName: `${ROOT_PATH}/${packageName}.js`,
+        vendorIndexFileName: `${VENDOR_CJS_PATH}/${npmAlias}/src/index.js`,
+        vendorPath: `${VENDOR_CJS_PATH}/${npmAlias}`,
+        nodeModulesPath: `${ROOT_NODE_MODULES_PATH}/${npmAlias}`,
+      };
+
+      all[packageName] = pkg;
+      return all;
+    },
+    {},
+  );
+
+  return result;
+};
+
+export const parsedVendorPkgsMap = parseVendorPkgs(packageJson.dependencies);
+
+const getLicenseUrl = (pkgJson: PackageJson) =>
+  `${pkgJson.repository.url.replace(/\.git$/, '')}/blob/main/LICENSE`;
+
+/** Generates the content of the vendored ESM package. */
+export function getESMContent(pkgJson: PackageJson, pkg: VendoredPkg) {
+  return `/**
+ * \`@visx/vendor/${pkg.packageName}\` (ESM)
+ * See upstream license: ${getLicenseUrl(pkgJson)}
+ *
+ * This ESM package re-exports the underlying installed dependencies of 
+ * \`node_modules/${pkg.packageName}\` (aliased as \`${pkg.npmAlias}\`)
+ */
+export * from '${pkg.npmAlias}';`;
+}
+
+/** Generates the content of the vendored CJS package. */
+export function getCJSContent(pkgJson: PackageJson, pkg: VendoredPkg) {
+  return `/**
+ * \`@visx/vendor/${pkg.packageName}\` (CommonJS)
+ * See upstream license: ${getLicenseUrl(pkgJson)}
+ *
+ * This CJS package exports transpiled vendor files in \`${VENDOR_CJS_DIR}${pkg.npmAlias}\`
+ */
+module.exports = require('../${VENDOR_CJS_DIR}${pkg.npmAlias}/src/index.js');`;
+}
+
+/** Generates the content of the root index file which points to CJS. */
+export function getIndexContent(pkgJson: PackageJson, pkg: VendoredPkg) {
+  return `/**
+ * \`@visx/vendor/${pkg.packageName}\` (CommonJS)
+ * See upstream license: ${getLicenseUrl(pkgJson)}
+ *
+ * This file only exists for tooling that doesn't work yet with package.json:exports
+ * by proxying through the CommonJS version.
+ */
+module.exports = require('./${VENDOR_CJS_DIR}${pkg.npmAlias}/src/index.js');`;
+}
+
+/** Generates the content of the vendored TS types. */
+export function getTSContent(pkg: VendoredPkg) {
+  return `/** \`@visx/vendor/${pkg.packageName.replace('@types/', '')}\` (TypeScript) 
+ *
+ * Re-exports the types from \`${pkg.packageName}\` 
+ */
+export * from '${pkg.npmAlias}';`;
+}
+
+// note: this is how we pass these dynamic variables into the
+// babel config. it's not great but babel config files can't easily
+// import from TS module files like this
+process.env.VENDOR_CJS_PATH = VENDOR_CJS_PATH;
```

Comment:
noting I spent a TON of time trying to figure out how to leverage these inside the `babel.config.js` file ... you can't import the values from an ESM module like this, this was the cleanest way I came up with